src/app/dataset-view/additional-components/data-component/data-component.ts

```diff
@@ -27,14 +27,17 @@ import { sqlEditorOptions } from "src/app/dataset-block/metadata-block/component
 })
 export class DataComponent extends BaseComponent implements OnInit {
     @Input() public datasetBasics?: DatasetBasicsFragment;
-    @Output() public runSQLRequestEmit = new EventEmitter<string>();
+    @Output() public runSQLRequestEmit = new EventEmitter<DatasetRequestBySql>();
     public sqlEditorOptions = sqlEditorOptions;
     public savedQueries = DataTabValues.savedQueries;
     public sqlRequestCode = `select\n  *\nfrom `;
 
     public sqlErrorMarker: MaybeNull<string> = null;
     public currentSchema: MaybeNull<DatasetSchema> = null;
     public currentData: DataRow[] = [];
+    public defaultLimit = AppValues.SQL_QUERY_LIMIT;
+    public queryLimits: number[] = [AppValues.SQL_QUERY_LIMIT, 100, 200, 500];
```

Comment:
Maybe a formula based on `AppValues.SQL_QUERY_LIMIT` * 2, * 4. ...

Reply:
This functionality is outdated